packages/core/upload/admin/src/components/TableList/index.js

```diff
@@ -40,12 +54,46 @@ export const TableList = ({
               }
             />
           </Th>
-          {tableHeaders.map(({ metadatas, key }) => {
+          {tableHeaders.map(({ metadatas: { label, isSortable }, name, key }) => {
+            const isSorted = sortBy === name;
+            const isUp = sortOrder === 'ASC';
+            const tableHeaderLabel = formatMessage(label);
+            const sortLabel = formatMessage(
+              { id: 'list-table-header-sort', defaultMessage: 'Sort on {label}' },
+              { label: tableHeaderLabel }
+            );
+
             return (
-              <Th key={key}>
-                <Typography textColor="neutral600" variant="sigma">
-                  {formatMessage(metadatas.label)}
-                </Typography>
+              <Th
+                action={
+                  isSorted && (
+                    <IconButton
+                      label={sortLabel}
+                      onClick={() => handleClickSort(isSorted, name)}
+                      icon={isUp ? <CarretUp /> : <CarretDown />}
```

Comment:
Worth passing a children instead? wdyt?

Reply:
why not yes! just out of curiosity, is there a noticeable difference between props/children? for better reading?